cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -369,11 +369,11 @@ func StartControllers(s *options.CMServer, kubeconfig *restclient.Config, rootCl
 	// Find the list of namespaced resources via discovery that the namespace controller must manage
 	namespaceKubeClient := client("namespace-controller")
 	namespaceClientPool := dynamic.NewClientPool(restclient.AddUserAgent(kubeconfig, "namespace-controller"), restMapper, dynamic.LegacyAPIPathResolverFunc)
-	groupVersionResources, err := namespaceKubeClient.Discovery().ServerPreferredNamespacedResources()
+	deletableGroupVersionResources, err := namespaceKubeClient.Discovery().ServerPreferredNamespacedResourcesWithVerbs("delete")
```

Comment:
Do we have other cases in the API where `nil` and empty list have distinct meanings?

I only know a number of cases like https://github.com/sttts/kubernetes/blob/b22ccc6780c843b29dbd0da6e44f20c56870b09b/pkg/apis/extensions/types.go#L836-L836 where empty means "all".

To preserve the distinction we can do
```golang
SupportedVerbs *struct{
  Verbs []string
}
```